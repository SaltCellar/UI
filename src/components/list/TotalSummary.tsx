import { Title } from '@patternfly/react-core';
import { css } from '@patternfly/react-styles';
import React from 'react';
import { styles } from './header.styles';

interface TotalSummaryProps {
  value: string | number;
  totalLabel: string;
  dateLabel: string;
}

export const TotalSummary: React.SFC<TotalSummaryProps> = ({
  value,
  totalLabel,
  dateLabel,
}) => (
  <div className={css(styles.total)}>
    <Title className={css(styles.totalValue)} size="4xl">
      {value}
    </Title>
    <div className={css(styles.totalLabel)}>
      <div className={css(styles.totalLabelUnit)}>{totalLabel}</div>
      <div className={css(styles.totalLabelDate)}>{dateLabel}</div>
    </div>
  </div>
);
