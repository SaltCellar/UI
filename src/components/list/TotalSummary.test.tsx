import { Title } from '@patternfly/react-core';
import * as emotion from 'emotion';
import { shallow } from 'enzyme';
import { createSerializer } from 'jest-emotion';
import React from 'react';
import { TotalSummary } from './TotalSummary';

expect.addSnapshotSerializer(createSerializer(emotion));

test('render total summary', () => {
  const view = shallow(
    <TotalSummary
      value="$27.12"
      totalLabel={'Total Cost'}
      dateLabel={'5 weeks ago'}
    />
  );
  expect(view).toMatchSnapshot();
});
