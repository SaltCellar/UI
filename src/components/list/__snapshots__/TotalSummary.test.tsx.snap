// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render total summary 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 {
  margin-top: 0;
  margin-bottom: 0;
  margin-right: var(--pf-global--spacer--md);
}

.emotion-1 {
  font-size: 0.875rem;
  color: var(--pf-global--Color--100);
}

.emotion-2 {
  font-size: 0.875rem;
  color: var(--pf-global--Color--200);
}

<div
  className="emotion-4"
>
  <Title
    className="emotion-0"
    size="4xl"
  >
    $27.12
  </Title>
  <div
    className="emotion-3"
  >
    <div
      className="emotion-1"
    >
      Total Cost
    </div>
    <div
      className="emotion-2"
    >
      5 weeks ago
    </div>
  </div>
</div>
`;
